controller/space_template_blackbox_test.go

```diff
@@ -0,0 +1,378 @@
+	return svc, NewSpaceTemplateController(svc, db, cfg)
+}
+
+func unsecuredSpaceTemplateController(db *gormapplication.GormDB, cfg *config.ConfigurationData) (*goa.Service, *SpaceTemplateController) {
+	svc := goa.New("SpaceTemplate-Service")
+	return svc, NewSpaceTemplateController(svc, db, cfg)
+}
+
+func (s *TestSpaceTemplateSuite) SecuredController() (*goa.Service, *SpaceTemplateController) {
+	return securedSpaceTemplateController(s.db, s.Configuration)
+}
+
+func (s *TestSpaceTemplateSuite) UnSecuredController() (*goa.Service, *SpaceTemplateController) {
+	return unsecuredSpaceTemplateController(s.db, s.Configuration)
+}
+
+func (s *TestSpaceTemplateSuite) createMinimumSpaceTemplate(t *testing.T) *app.SpaceTemplateSingle {
+	return createMinimumSpaceTemplate(t, s.db, s.Configuration)
+}
+
+func createMinimumSpaceTemplate(t *testing.T, db *gormapplication.GormDB, cfg *config.ConfigurationData) *app.SpaceTemplateSingle {
+	// given
+	var res *app.SpaceTemplateSingle
+	var appSpaceTemplate *app.SpaceTemplate
+	svc, ctrl := securedSpaceTemplateController(db, cfg)
+	req := &goa.RequestData{
+		Request: &http.Request{Host: "api.service.domain.org"},
+	}
+	desc := "a random description"
+	spaceTemplate := spacetemplate.SpaceTemplate{
+		ID:          uuid.NewV4(),
+		Name:        uuid.NewV4().String(), // in order to create a random space template name
+		Description: &desc,
+		Template:    spacetemplate.GetValidEmptyTemplate(),
+	}
+	// when
+	application.Transactional(db, func(appl application.Application) error {
+		appSpaceTemplate = ConvertSpaceTemplate(appl, req, spaceTemplate)
+		payload := &app.CreateSpaceTemplatePayload{
+			Data: appSpaceTemplate,
+		}
+		_, res = test.CreateSpaceTemplateCreated(t, svc.Context, svc, ctrl, payload)
+		return nil
+	})
+	// then
+	require.NotNil(t, res)
+	require.NotNil(t, res.Data)
+	require.NotNil(t, res.Data.ID)
+	require.Equal(t, spaceTemplate.ID, *res.Data.ID)
+
+	require.NotNil(t, res.Data.Attributes)
+	require.NotNil(t, res.Data.Attributes.Name)
+	require.Equal(t, spaceTemplate.Name, *res.Data.Attributes.Name)
+
+	require.NotNil(t, *res.Data.Attributes.Description)
+	require.Equal(t, *spaceTemplate.Description, *res.Data.Attributes.Description)
+
+	require.NotNil(t, *res.Data.Attributes.Template)
+	require.Equal(t, *appSpaceTemplate.Attributes.Template, *res.Data.Attributes.Template)
+	return res
+}
+
+func (s *TestSpaceTemplateSuite) TestSpaceTemplate_Create() {
+	req := &goa.RequestData{
+		Request: &http.Request{Host: "api.service.domain.org"},
+	}
+
+	s.T().Run("invalid YAML template", func(t *testing.T) {
+		t.Parallel()
+		// given
+		svc, ctrl := s.SecuredController()
+		spaceTemplate := spacetemplate.SpaceTemplate{
+			Name:     uuid.NewV4().String(), // in order to create a random space template name
+			Template: "this is not a valid template ;)",
+		}
+		var jerr *app.JSONAPIErrors
+		// when
+		application.Transactional(s.db, func(appl application.Application) error {
```

Comment:
Sorry about that. I totally forgot that `s.db` (which is a `gormsupport.Application`) was implementing `application.Application`. Fixed in 5ec3aa9e112108f1ffc2d522b1f6a4b0886eaad6